sympy/codegen/ast.py

```diff
@@ -188,7 +188,7 @@ def _get_constructor(cls, attr):
     @classmethod
     def _construct(cls, attr, arg):
         """ Construct an attribute value from argument passed to ``__new__()``. """
-        if arg is None:
+        if arg == None: # Must be "== None", cannot be "is None"
```

Comment:
I was under the impression that is it best practices to use `is` for checks against the builtin `None`. What is the reasoning behind these changes?

Reply:
One note on this: https://docs.quantifiedcode.com/python-anti-patterns/readability/comparison_to_none.html

PEP 8 recommends is.

Reply:
The original reasoning was to use `is None` (see https://github.com/sympy/sympy/pull/16051/commits/4a54999df2eaed7b1f2c4d07f4fc2fd20c6aee19 where it was changed to that). However, as the test then failed I had to change it back. Not clear to me why it happens though (I haven't tries to figure it out either, just changed it back to make it work as before).

Reply:
@moorepants and @jksuom it is actually a bit demotivating to get comments like this and encouragements of those comments when the EXACT purpose of the PR was to improve the code quality based on PEP8. If you would have checked all commits or just read the commit messages, that should have been obvious to you. In addition, it is very, very easy to change this single line in your own local copy and see if it works or if there actually might have been a reason to revert it.

Reply:
Having just clicked the link from the email it took me to a diff showing only the changes from the last two commits which made it look like you were changing this from `is None` to `== None`. I can understand the confusion there. In any case I'm interested to know why it can't be is None. In what situation can is there an object equal to None that is not None itself?

Reply:
> If you would have checked all commits or just read the commit messages, that should have been obvious to you. In addition, it is very, very easy to change this single line in your own local copy and see if it works or if there actually might have been a reason to revert it.

I have no intention to demotivate. You also presume we all have vast amount of time to check every detail as you suggest. I've been reviewing code for just about 10 years on this project and and there are typical things that I notice as unusual. I certainly may have made a mistake in the review, but when I clicked the notification it did indicate a change from `is` to `==` as @oscarbenjamin mentions. This is a typical thing I may comment on.  I'm happy to be educated as to why it is useful in this case. We value your pull requests and I am happy to improve my review style based on your feedback.

Reply:
It seems to me that `==` is correct in this case. The reason is that `ast` defines a `NoneToken` class whose instances compare equal to `None`. So `arg == None` is an abbreviation of `arg is None or isinstance(arg, NoneToken)`.

Reply:
Thanks for the explanation @jksuom. That seems a bit of an antipattern though: the reason for saying that `is None` is preferred is that there shouldn't be any reason to create alternative None objects.

Does this line actually need to handle both cases or can it just be `isinstance(arg, NoneToken)`?

Reply:
> Does this line actually need to handle both cases or can it just be `isinstance(arg, NoneToken)`?

I just tried this and got a test failure from `test_Declaration` so I guess it does need to be `== None`.

Reply:
@moorepants I fully understand that you have limited time and I appreciate everyone that is reviewing. However, there is clearly a challenge when just reviewing incremental commits and, to some extent, assuming that the change is "stupid". Sometimes it is not and then it may happen that the author gets the impression that the assumption of the change also holds for the author...

But I guess that being relatively new, it is something I have to (and can, even though I needed to let off some steam) live with. In ten years, the expectations of my code quality is probably a bit higher so that the assumption is that it probably is needed.

Reply:
So, well, trying to be a bit constructive: maybe it is a good idea (for all of us who reviews) to check out the whole PR in case there is something that we find "stupid" (in lack of a better word) in an incremental commit, to actually confirm that it is.